src/libos/src/process/do_spawn/init_stack.rs

```diff
@@ -58,6 +58,22 @@ pub fn do_init(
     let stack_buf = unsafe { StackBuf::new(stack_top, init_area_size)? };
     let envp_cloned = clone_cstrings_on_stack(&stack_buf, envp)?;
     let argv_cloned = clone_cstrings_on_stack(&stack_buf, argv)?;
+    // Put 8 byte to make the postion of stack 8-byte aligned
+    stack_buf.put(0 as u64);
+    let current_pos = stack_buf.get_pos();
+    let to_alloc_size = {
+        let auxtbl_size = calc_auxtbl_size_on_stack(auxtbl);
+        let envp_size = calc_cstrptrs_size_on_stack(&envp_cloned);
+        let argv_size = calc_cstrptrs_size_on_stack(&argv_cloned);
+        let argc_size = mem::size_of::<u64>();
+        auxtbl_size + envp_size + argv_size + argc_size
+    };
+    // Libc ABI requires 16-byte alignment of the stack entrypoint.
+    // Current postion of the stack is 8-byte aligned already, insert 8 byte
+    // to meet the requirement if necessary.
+    if (current_pos - to_alloc_size) % 16 != 0 {
```

Comment:
Is it better to add a function in StackBuf to adjust the alignment?

Reply:
@liqinggd The piece of code inserted here is very readable. I like it. I think it can be improved as @guzongmin points out: extract the inserted code as a separate function and give it a very readable name. It could be `adjust_alignment`. Maybe you can figure out something better.

Reply:
Sure, I have refined the commit